bpf/include/bpf/settings.h

```diff
@@ -0,0 +1,29 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
+/* Copyright Authors of Cilium */
+
+#ifndef __LIB_SETTINGS_H_
+#define __LIB_SETTINGS_H_
+
+#include <bpf/compiler.h>
+
+#define SETTING_HARDCODED(name, val) \
+	static __always_inline unsigned long long setting_##name(void)\
+	{ \
+		return val; \
+	} \
+
+#define SETTING(name) \
```

Comment:
Bikeshedding time: could we find a term other than 'setting'? It's a conjugated verb, so it's a little unnatural to read. `CONFIG` would be another possible choice, and is a little shorter to type.